src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,428 @@ class Guides::Database::Querying < GuideAction
 
-    ### Customizing how associations are preloaded
+    ### Avg / Sum
 
-    Sometimes you want to order preloads, or add where clauses. To do this, use
-    the `preload_{{ association_name }}` method on the query and pass a query
-    object for the association.
+    `SELECT AVG(users.age) FROM users`
 
     ```crystal
-    PostQuery.new.preload_comments(CommentQuery.new.published(true))
+    UserQuery.new.select_average(:age)
     ```
 
-    This is also how you would do nested preloads:
+    `SELECT SUM(users.age) FROM users`
 
     ```crystal
-    # Preload the post's comments, and the comment's author
-    PostQuery.new.preload_comments(CommentQuery.new.preload_author)
+    UserQuery.new.select_sum(:age)
     ```
 
-    > Note that you can only pass query objects to `preload` if the association
-    is defined, otherwise you will get a type error.
+    ### Min / Max
+
+    `SELECT MIN(users.age) FROM users`
 
-    ## Loading associations without preloading
+    ```crystal
+    UserQuery.new.select_min(:age)
+    ```
 
-    Sometimes you have a single model and don't need to preload items. Or maybe
-    you *can't* preload because the model record is already loaded. In those cases
-    you can use the association name with `!`:
+    `SELECT MAX(users.age) FROM users`
 
     ```crystal
-    post = PostQuery.first
-    # Returns the associated author and does not trigger a preload error
-    post.author!
+    UserQuery.new.select_max(:age)
     ```
 
-    ## Joining and querying associations
+    ## Associations and Joins
 
-    Let's say you have a `Post` model with `has many comments: Comment`, the
-    `Comment` model has a `rating` column that is an integer, and the highest rating
-    is 5 and lowest is 1.
+    ### Joins
 
-    If we want to get all posts that have at least 1 highly rated comment, we
-    could do something like this:
+    ## Preloading
+
+    ### without preloading
+
+    ## No results
+
+    Avram gives you a `none` method to return no results. This can be helpful when under
+    certain conditions you want the resultd to be empty.
 
     ```crystal
-    class PostQuery < Post::BaseQuery
-      def with_highly_rated_comments
-        # Join the comments
-        join_comments
+    UserQuery.new.none
+    ```
 
-        # Get posts with at least one comment with rating greater than or equal to 4
-        comments { |comment_query| comment_query.rating.gte(4) }
+    > This method does not return an empty array immediately. You can still chain other query methods,
+    > but it will always return no records. For example: `UserQuery.new.none.first` will never return a result
 
-        # Or do the same thing with the block shorthand syntax
-        # See "Short one-argument syntax" in https://crystal-lang.org/docs/syntax_and_semantics/blocks_and_procs.html
-        comments(&.rating.gte(4))
+    ## Named Scopes
+
+    Generally it’s best to name your query methods so they are easy to understand and reuse across your app.
```

Comment:
```suggestion
    Chaining multiple query methods can be hard to read, tedious, and error prone. If you are making a complex query more than once, or want to give a query a label, named scopes are a great alternative.
```

Sorry didn't want to just clobber your thing here, I just thought it could use a bit more explanation of _why_ named scoped should be used.

Reply:
haha. no worries! Thanks for the help!